### ISSUE TITLE
Replace modal screens with modals from `@react-navigation`

### ISSUE THREAD

**Issue by chrispader** [Expensify/App#53493]
P/S Slack thread: [expensify.slack.com/archives/C01GTK53T8Q/p1733938583809829](https://expensify.slack.com/archives/C01GTK53T8Q/p1733938583809829)

### Problem

As discussed in [this issue](https://github.com/Expensify/App/pull/53475#issue-2715390810) and this [Slack thread](https://expensify.slack.com/archives/C083CKCMWRE/p1733241544952329?thread_ts=1733228986.361849&cid=C083CKCMWRE) we should replace modal screens (navigation screens/routes, that only render a modal) and modals that stay open when the containing screen gets unmounted.

The current modals use `react-native-modal` which uses RN internal Modal under the hood. Using these modals with `@react-navigation/native-stack` poses issues with animations and navigation, as these modals cannot animate in/out, due to the screen mounting/unmounting at the same time. Additionally, the animations will not look the same as the rest of the screen animations, which doesn't look very nice.

### Solution

Replace the following types of modals with modal (screens) from `@react-navigation`

- screens/routes that only render a modal
- modals on screens, that are navigated back from while the modal is still open or animating out at the same time

cc @mountiny 

---

# List of components and screens to migrate to `@react-navigation` modal screens

Tracking my progress on an exhaustive list of components and screens that can/should be migrated to modal screens.

The list of components/screens (and other files) are triaged/prioritized as follows:
(File names are listed without file ending, usually `.tsx`)

- 🔴 **HIGH**: Should definitely be migrated, due to current issues with animations and navigation.

  These modals are already causing issues with animations, because

  - they are **animating in while a new screen is still mounting**
  - they are **still open/displayed while the old screen unmounting**

  causing no or cut off animations.

- 🟡 **MEDIUM**: Not causing any issues at the moment, but still worth to migrate to modal screens since the behave "like a screen", to achieve consistent animations and generally improve navigation. (gestures, native feel)
 
  This includes modals that are either
  - **animated in from right-to-left** (like a screen) or
  - that are part of another screen with **`animation: none`**,
  - etc.

- 🟢 **LOW**: These screens are **animated in from bottom-to-top** like actual modals and are never (un-)mounted while at the same time as the screen. Migrating these modals to modal screens (and extra navigation routes) would make animations more consistent with the other screens, but it is not necessary and might not be worth the effort.

  Using a custom modal implementation or something like `gorhom/react-native-bottom-sheet` probably makes more sense.

## Base components to (potentially) re-write

<details>
<summary>BaseModal</summary>

- Affected modals:
  - `TransactionStartDateSelectorModal`
  - `PaymentCardCurrencyModal`
  - `AmountSelectorModal`
  - `AttachmentModal`
  - `AvatarCropModal`
  - `CategorySelectorModal`
  - `ConfirmModal`
  - `CountrySelectorModal`
  - `YearPickerModal`
  - `DecisionModal`
  - `FeatureTrainingModal`
  - `PushRowModal`
  - `RequireTwoFactorAuthenticationModal`
  - `SearchRouterModal`
  - `StateSelectorModal`
  - `TestToolsModal`
  - `TextSelectorModal`
  - `ValueSelectorModal`
  - `BusinessTypeSelectorModal`
  - `NetSuiteCustomListSelectorModal`
  - `ExpenseLimitTypeSelectorModal`
  - `TransactionStartDateSelectorModal`
  - `UnitSelectorModal`
  - `WorkspaceMemberDetailsRoleSelectionModal`
  - `InitialListValueSelectorModal`
  - `TypeSelectorModal`
- Pages:
  - `EditReportFieldDate`
  - `SetDatePage`
  - `DateOfBirthPage`
  - `SearchSelectedNarrow`
- Components:
  - `SearchDateFilterBase`
  - `DateOfBirthStep`
  - `DebugDetailsDateTimePickerPage`
  - `AdditionalDetailsStep`
  - `IncorporationDateBusiness`
  - `IOURequestStepDate`

</details>

<details>
<summary>Popover (uses Modal)</summary>

- Comment: Uses `Modal` under the hood
- Components:
  - `EmojiPicker`
  - `PopoverMenu`
  - `PopoverWithMeasuredContent`
  - `PopoverReportActionContextMenu`
  - `BasePopoverReactionList`
  - `ProcessMoneyReportHoldMenu`
  - `WorkspaceCardsListLabel`
  - `AccountSwitcher`
  - `AddPaymentMethodMenu`
  - `AvatarWithImagePicker`
  - `AttachmentPickerWithMenu`

</details>

## Triaging

Modals only visible on native platforms (Android & iOS) are marked with 📱.

### 🔴 **HIGH**: Modals causing issues at the moment

Modals animating in from the bottom are marked with ⬆️, modals animating in from the right with ⬅️

<details>
<summary>AttachmentModal ⬅️ (screen recordings attached)</summary>

- Comment: Add one or many extra routes?
- Components:
  - `AvatarWithImagePicker`
  - `ReportAvatar`
  - `TransactionReceiptPage`
  - `ReportActionCompose`
  - `ReportAttachments`
  - `ProfileAvatar`
  - `WorkspaceAvatar`

- iOS: Animating in not working, because the modal is animating at the same time, as a new screen with `animation: none` is mounting.

  https://github.com/user-attachments/assets/2f46c232-76e9-4279-b685-d6ede5aba40d

- Android: Animation out is not working, probably due to the screen unmounting before the modal can animate out

  https://github.com/user-attachments/assets/7118a425-6ead-4f11-92a9-5ab5e69d7ca1

</details>

<details>
<summary>ConfirmModal ⬆️ (screen recordings attached)</summary>

- Comment: Not all `ConfirmModal`s are causing issues
- Screens, where we navigate back while also animating the modal out:
  - `TagSettingsPage` (Remove tag)
  - `WorkspaceEditTaxPage` (Remove tax)
  - `CategorySettingsPage` (Remove category)
  - `PolicyDistanceRateDetailsPage` (Remove distance rate)

- iOS: Modal not able to animate out, because the screen that wraps the modal is already unmounting:

  https://github.com/user-attachments/assets/bba04c1e-4164-4e9b-a5e1-3272fd5e22bd

- Android: Same issue, though less obvious:

  https://github.com/user-attachments/assets/03d8c066-e1fe-40e8-a2f8-2766ab8c7b03

</details>

### 🟡 **MEDIUM**: Screen-like modals - Animating in from right-to-left

<details>
<summary>SearchRouterModal</summary>

- Pages:
  - `AuthScreens`

</details>

<details>
<summary>AmountSelectorModal -> AmountPicker</summary>

- Pages:
  - `Form`
  - `WorkspaceCreateTaxPage`

</details>

<details>
<summary>AvatarCropModal -> AvatarWithImagePicker</summary>

- Pages:
  - `NewChatConfirmPage`
  - `ReportDetailsPage`
  - `ProfilePage`
  - `WorkspaceProfilePage`

</details>

<details>
<summary>CategorySelectorModal</summary>

- Pages:
  - `WorkspaceCategoriesSettingsPage`
- Components:
  - `CategorySelector`
    - Pages:
      - `PolicyDistanceRatesSettingsPage`
      - `WorkspacePerDiemSettingsPage` (unused?)

</details>

<details>
<summary>YearPickerModal -> CalendarPicker -> DatePicker</summary>

- _Comment: Didn't test all flows_
- Pages:
  - `SearchDateFilterBase`
  - `DateOfBirthStep`
  - `DebugDetailsDateTimePickerPage`
  - `EditReportFieldDate`
  - `AdditionalDetailsStep`
  - `IncorporationDateBusiness`
  - `IOURequestStepDate`
  - `SetDatePage`
  - `DateOfBirthPage`
- Components:
  - `TransactionStartDateSelectorModal`

</details>

<details>
<summary>TextSelectorModal -> TextPicker</summary>

- Pages:
  - `CreateReportFieldsPage`
  - `WorkspaceCreateTaxPage`
- Components:
  - `Form`

</details>

<details>
<summary>ValidateCodeActionModal</summary>

- Pages:
  - `MissingPersonalDetailsContent` (multiple occurrences)
  - `BankAccountStep`
  - `ContactMethodDetailsPage` (multiple occurrences)
  - `NewContactMethodPage`
  - `DelegateMagicCodeModal`
  - `CodesStep`
  - `BaseGetPhysicalCard`
  - `ExpensifyCardPage`
  - `ReportCardLostPage`
  - `ReportVirtualCardFraudPage`
  - `VerifyAccountPage`
  - `ConfirmationStep`

</details>

<details>
<summary>ValueSelectorModal -> ValuePicker</summary>

- Pages:
  - `WorkspaceNewRoomPage`
- Components:
  - `Form` (multiple occurrences)

</details>

<details>
<summary>FeatureTrainingModal</summary>

- Pages:
  - `TrackTrainingPage`
- Components:
  - `ExplanationModal`
  - `MigratedUserWelcomeModal`
  - `OnboardingWelcomeVideo`

</details>

<details>
<summary>PaymentCardCurrencyModal</summary>

</details>

<details>
<summary>ConnectToQuickbooksOnlineFlow 📱</summary>

</details>

<details>
<summary>ConnectToXeroFlow 📱</summary>

</details>

<details>
<summary>CardAuthenticationModal</summary>

- Pages:
  - `PaymentCard`
  - `WorkspaceOwnerChangeWrapperPage`

</details>

<details>
<summary>NetSuiteCustomListSelectorModal -> NetSuiteCustomListPicker</summary>

- Pages:
  - `ChooseCustomListStep`

</details>

<details>
<summary>ExpenseLimitTypeSelectorModal -> ExpenseLimitTypeSelector</summary>

- Pages:
  - `CategoryFlagAmountsOverPage`

</details>

<details>
<summary>TransactionStartDateSelectorModal</summary>

- Components:
  - `TransactionStartDateStep`

</details>

<details>
<summary>UnitSelectorModal -> UnitSelector</summary>

- Pages:
  - `PolicyDistanceRatesSettingsPage`

</details>

<details>
<summary>WorkspaceMemberDetailsRoleSelectionModal</summary>

- Pages:
  - `WorkspaceMemberDetailsPage` (multiple occurrences)

</details>

<details>
<summary>InitialListValueSelectorModal -> InitialListValueSelector</summary>

- Pages:
  - `CreateReportFieldsPage`

</details>

<details>
<summary>TypeSelectorModal -> TypeSelector</summary>

- Pages:
  - `CreateReportFieldsPage`

</details>

### 🟢 **LOW**: Contained modals - Only visible as part of the current screen, animating in from bottom-to-top

<details>
<summary>ConfirmModal</summary>

- Comment: We will need to decide about creating an extra route for every screen individually.
- Screens:
  - `EditReportFieldPage`
  - `ReportDetailsPage`
  - `ReportParticipantDetailsPage`
  - `ReportParticipantsPage`
  - `RoomMemberDetailsPage`
  - `RoomMembersPage`
  - `EmptySearchView`
  - `ConsolePage`
  - `InitialSettingsPage`
  - `ContactMethodDetailsPage`
  - `VisibilityPage`
  - `CloseAccountPage`
  - `SecuritySettingsPage`
  - `TroubleshootPage`
  - `WalletPage`
  - `WorkspaceInitialPage`
  - `WorkspaceMembersPage`
  - `WorkspaceMoreFeaturesPage`
  - `WorkspaceProfilePage`
  - `WorkspacesListPage`
  - `PolicyAccountingPage`
  - `SageIntacctEditUserDimensionsPage`
  - `NetSuiteImportCustomFieldView`
  - `ImportedCategoriesPage`
  - `WorkspaceCategoriesPage`
  - `WorkspaceCompanyCardDetailsPage`
  - `WorkspaceCompanyCardsSettingsPage`
  - `PolicyDistanceRatesPage`
  - `WorkspaceEditCardLimitPage`
  - `WorkspaceEditCardLimitTypePage`
  - `WorkspaceExpensifyCardDetailsPage`
  - `WorkspaceExpensifyCardPageEmptyState`
  - `WorkspaceInvoiceVBASection`
  - `ImportedMembersPage`
  - `WorkspaceMemberDetailsPage`
  - `ImportedPerDiemPage`
  - `WorkspacePerDiemDetailsPage`
  - `WorkspacePerDiemPage`
  - `ReportFieldsListValuesPage`
  - `ReportFieldsSettingsPage`
  - `ReportFieldsValueSettingsPage`
  - `WorkspaceReportFieldsPage`
  - `ImportedTagsPage`
  - `WorkspaceTagsPage`
  - `WorkspaceViewTagsPage`
  - `WorkspaceTaxesPage`
  - `WorkspaceWorkflowsPage`
  - `WorkspaceWorkflowsApprovalsEditPage`
- Components:
  - `Expensify`
  - `AccountSwitcher`
  - `AccountingConnectionConfirmationModal`
  - `AttachmentModal`
  - `ConfirmModal`
  - `DelegateNoAccessModal`
  - `FocusModeNotification`
  - `ImportSpreadsheet`
  - `LocationPermissionModal`
  - `MoneyReportHeader`
  - `ExportWithDropdownMenu`
  - `SearchPageHeader`
  - `SupportActionRestrictedModal`
  - `BaseUpdateAppModal`
  - `HeaderView`
  - `PopoverReportActionContextMenu`
  - `ReportDetailsExportPage`
  - `FloatingActionButtonAndPopover`
  - `IOURequestStepScan`
  - `IOURequestStepWaypoint`
  - `CardSection`
  - `WorkspaceResetBankAccountModal`
- Hooks:
  - `useDeleteSavedSearch`

</details>

<details>
<summary>DecisionModal</summary>

- Pages:
  - `SearchPageHeader`
  - `ReportDetailsPage` (multiple occurrences)
  - `WorkspaceMembersPage` (multiple occurrences)
  - `WorkspaceCategoriesPage` (multiple occurrences)
  - `WorkspaceTagsPage`
- Components:
  - `BaseImportOnyxState`
  - `ProcessMoneyReportHoldMenu`

</details>

<details>
<summary>SelectionListModal -> SelectionListWithModal</summary>

- Pages:
  - `Search`
  - `ReportParticipantsPage`
  - `RoomMembersPage`
  - `WorkspaceMembersPage`
  - `WorkspaceCategoriesPage`
  - `PolicyDistanceRatesPage`
  - `WorkspacePerDiemPage`
  - `ReportFieldsListValuesPage`
  - `WorkspaceReportFieldsPage`
  - `WorkspaceTagsPage`
  - `WorkspaceViewTagsPage`
  - `WorkspaceTaxesPage`

</details>

<details>
<summary>RequireTwoFactorAuthenticationModal</summary>

- Components:
  - `Expensify` (multiple occurrences)

</details>

<details>
<summary>TestToolsModal</summary>

- Pages:
  - `AuthScreens`

</details>

<details>
<summary>SearchSelectedNarrow -> SearchPageHeader</summary>

- Pages:
  - `SearchPage`
  - `SearchSelectionModaHeader`

</details>

### ❓ Unknown: Unable to test because the modal is either missing or not accessible

<details>
<summary>CountrySelectorModal -> CountryPicker -> Address</summary>

- Pages:
  - `MissingPersonalDetailsContent` -> `MissingPersonalDetails` (unused?)

</details>

<details>
<summary>StateSelectorModal -> StatePicker -> Address</summary>

- Components:
  - `MissingPersonalDetailsContent` -> `MissingPersonalDetails` (unused?)

</details>

<details>
<summary>PushRowModal -> PushRowWithModal (unused?)</summary>

- Pages:
  - `ReimburseAccountPage`
  - `AddressFormFields`
  - `BusinessType`
  - `IncorporationLocation`
  - `PaymentVolume`
  - `Confirmation`

</details>

<details>
<summary>BusinessTypeSelectorModal -> BusinessTypePicker</summary>

- Pages:
  - `ReimburseAccountPage`
  - `BusinessInfo`

</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @parasharrajat</details>

**Comment by mountiny** [comment]
@chrispader we will have to discuss this and make a proposal first

**Comment by mountiny** [comment]
No, we are holding this for a proposal/ discussion around it

**Comment by mountiny** [comment]
Made chris the owner of the issue

**Comment by Christinadobrzyn** [comment]
Just a heads up that I'm going to be ooo Dec 12 - 13th. Back on Monday 16th. I'm not going to assign this to a BZ teammate but if anything is urgent, please reach out to the team for a volunteer.

**Comment by chrispader** [comment]
I've just posted a P/S thread here: https://expensify.slack.com/archives/C01GTK53T8Q/p1733938583809829

**Comment by Christinadobrzyn** [comment]
Nice work on the P/S @chrispader - looks like it has lots of votes to move forward. Let us know what you're thinking for next steps!

**Comment by mountiny** [comment]
I think w can take this off hold, adding this to quality as low as I think it aims to improve UX.

@chrispader can you please start with a plan on how to tackle this before writing any code. Also would be great to get agreement from @adamgrzybowski @WoLewicki on the next steps and maybe even work together with them as this is change close to screens and navigation

**Comment by mountiny** [comment]
@chrispader Do you have any ETA for moving this issue ahead? Do you need to hand over some of the other tasks to someone else in Margelo maybe?

**Comment by chrispader** [comment]
> @chrispader Do you have any ETA for moving this issue ahead? Do you need to hand over some of the other tasks to someone else in Margelo maybe?

I'm going to work on a proposal for this until the end of the week.

In the proposal i will skim through the app and try to find all the screens where it pays off to migrate them to modal screens.

**Comment by chrispader** [comment]
WIP, last updated at <time>

---

# Modal migration to @react-navigation modal screens (Chris)

## HIGH: Modals causing issues, visible on (un-)mount

Replace with modal screen to fix issues with animations and/or navigation.

### Components

- `AttachmentModal`: One or many extra routes?

---

## MEDIUM: Separate modals

Potentially replace with extra route and screen to achieve consistent animation and improve navigation.

This includes modals that are either navigated in from right-to-left or mount immediately after navigating to a different screen.

### Components

- `AmountPicker`: Is this always a separate screen?

---

## LOW: Contained modal - No need to create extra route (modal animation)

Can also be replaced by an extra screen to improve animations, but the effort might not be worth.

### Components

**Comment by chrispader** [comment]
# Proposal

Triage all screens and components that contain modals into strictly necessary, beneficial and potential but not necessary issues.

Once we've triaged all modals in screens and components into the three priorities/categories (described in the issue description) we can start migrating those components to use modal screens from `@react-navigation`. The migration will look somewhat like this:

1. Extract the modal (screen) content to a separate React component, if not already done so. Think about all the navigation props that are needed for the modal screen, to behave like before.
2. Create one or many new routes for each case in any flow, where the modal appears. e.g. the `AmountPicker` modal in the workspace taxes flow as something like `/workspaces/wallet/add/amount`.
3. Replace any usages of modals by (re-)navigating to the new screen with the appropriate props.
4. Test that functionality behaves the same as before.
5. (Potentially test direct navigation linking to the modal, if it can be reached through a URL directly)

**Comment by chrispader** [comment]
I've now updated the list of modals to the extent, that it includes all modals used in the app (that i could find through "smart" file search in VS Code and using AI Tools.

Lots of the files (modal components, other components, pages, etc.) listed under "TODO" appear more than once and might not need to be handled individually.

My next step would be to skim through the app (both in the web version and on iOS/Android) and check all of these modals and triage them by the described categories.

**Comment by Christinadobrzyn** [comment]
Awesome! Should I add you to this issue @parasharrajat - not sure if there's a different process for C+ on a new project.

**Comment by Christinadobrzyn** [comment]
No I think that's fine @parasharrajat! Thanks for checking though!

**Comment by chrispader** [comment]
I've just finished triaging all the modals in the codebase. (Hopefully i found all 😅)

For each of the modals, you can find a detailed list of all the screens and components that use it. The different modals are prioritized into the three described categories, 🔴HIGH, 🟡 MEDIUM, and 🟢 LOW.

The result from the triaging process now opens the following questions that need to be answered, before we want to start migrating the modals:

- Do we only want to fix **🔥 HIGH** priority modals or also other "screen-like" modals, that don't have consistent animations, gestures and usual screen navigation behavior?
- Do we want to work on all of this in one PR that migrates all modals or should we split up the workload into multiple PRs (and multiple assignees)?
-  We'll probably have to re-write the `BaseModal`, `Popover` and maybe other "base components", to work differently on web vs. native. Do you want me to work on this in a first "major PR", that the other PRs should hold on?

**Comment by BartoszGrajdek** [comment]
Hey Chris, thanks for mentioning me here! 🙌🏻  Your plan looks solid, and I really don't have much to add. 🪨 

Just a quick recap on what I've been up to (you've probably already seen it, since you linked it in your Slack message):
 
I'm currently wrapping up a PR for some enhancements to modals ([here's the issue](https://github.com/Expensify/App/issues/49354)). Our main goal was to swap out `react-native-animatable` with reanimated in `react-native-modal` library, targeting all modals in E/App across both _native_ and _mWeb_ platforms to improve the animation performance.

We decided to roll out these improvements starting with the **bottom docked modals**, which are currently on your **🟢 LOW** priority list. I understand you'll be handling the other modal types. So, I'll hold off on those unless plans change, or you decide that any other modals should stick to the previous solution – just let me know if they do! 👀 

I'm planning to raise a PR later this week. I'll tag you there so that you can take a look at it - there are some really small changes in `BaseModal`, but nothing that should worry you.

Also, you mentioned a bug in the **🔴 HIGH** priority -> `ConfirmModal`. Good news – that issue is resolved with our new setup using layout animations, which allows us to execute a callback immediately after an animation takes place, even if the modal unmounts. I've attached a video below demonstrating the new flow. Please take a look! 😄 

<details>
<summary>ConfirmModal Demo</summary>

https://github.com/user-attachments/assets/c0363cb2-9b0f-4784-a8cb-19723e577107

</details>

Let me know if you have any questions - I'm happy to help! 😊

**Comment by mountiny** [comment]
@chrispader What are your next steps?

**Comment by chrispader** [comment]
> [@chrispader](https://github.com/chrispader) What are your next steps?

I'm going to start working on a holistic change for the `AttachmentModal` specifically this week. Once i've migrated the `AttachmentModal`, i can start wokring on the other 🟡 modals

**Comment by chrispader** [comment]
No update yet. Going to continue working on this PR this week (especially a fix for the `AttachmentModal` component)

**Comment by jjcoffee** [comment]
@chrispader Could you confirm if your work will fix https://github.com/Expensify/App/issues/55537?

**Comment by mountiny** [comment]
@chrispader Can you please leave an update with the current progress and next steps? thanks!

**Comment by chrispader** [comment]
@mountiny @parasharrajat sorry for the delay on this one.

This is what i did so far:

-  For the `AttachmentModal` we'll actually want to transform this into a "regular" screen, rather than a modal screen on native platforms, since we're already treating it like any other screen (right-to-left).
  - I've already made that work, just need to finalize the PR. Basically we're using an the actual screen animations and presentation on native platforms (iOS/Android), while still preserving the "popover-like" appearance on web.
  - WIP PR: https://github.com/Expensify/App/pull/56219
- My second PR will add the necessary tools that current modals (e.g. `ConfirmModal`) will need to continue functioning as a separate screen + route.
  - Haven't set up a PR yet, but i already tried some changes. Will set up a PR for that soon!

**Comment by mountiny** [comment]
Thanks!

**Comment by chrispader** [comment]
The [first PR fixing all `AttachmentModal`s](https://github.com/Expensify/App/pull/56219) is done and in review.

I will soon prepare a PR that includes the necessary changes to base components and after that, apply it to a modal at a time in separate PRs. I'll start with migrating just one modal, and then we can see if it's worth the effort.

I'm pretty busy with improving bottom safe area padding across the app right now, so i'll probably not get to this until next week or so.

**Comment by Christinadobrzyn** [comment]
Hi @parasharrajat and @chrispader just checking in on this to see if there's an update. TY!

**Comment by Christinadobrzyn** [comment]
Amazing! thank you @chrispader!

**Comment by Christinadobrzyn** [comment]
Monitoring https://github.com/Expensify/App/pull/56219

**Comment by Christinadobrzyn** [comment]
Monitoring https://github.com/Expensify/App/pull/56219

**Comment by chrispader** [comment]
@Christinadobrzyn @mountiny once https://github.com/Expensify/App/pull/56219 is merged, let's keep this issue open though.

I'll continue migrating more modals when i find the time

**Comment by parasharrajat** [comment]
Update: Review continues on the first PR. Reviewed many related flows. Still need to test more on all platforms. 

Next update: Tomorrow.

**Comment by mountiny** [comment]
@chrispader We could probably define the tasks clearly for some C+ to take over, I think that will be better for us to do than you solely migrating it all. cc @staszekscp for visibility of this task

**Comment by chrispader** [comment]
> [@chrispader](https://github.com/chrispader) We could probably define the tasks clearly for some C+ to take over, I think that will be better for us to do than you solely migrating it all. cc [@staszekscp](https://github.com/staszekscp) for visibility of this task

Yes makes sense. I'll set up a list of instructions on what needs to be done per modal in order to migrate it to a `@react-navigation` modal. Then we can split up the work per modal or group of modals in the same flow.

After we merged https://github.com/Expensify/App/pull/56219 i'll extract some logic from the `AttachmentModal` and create a new "HybridModal" component or integrate it into our current `BaseModal` implementation, to be able to have a regular (Popover) modal on web and a proper modal screen on native.

This component can then be used by other devs to migrate all of the remaining modals marked with 🟡 . The work there should be pretty similar.

**Comment by chrispader** [comment]
@mountiny the ConfirmModal was the next one i would work on, because it causes issues at the moment and it might be trickier to handle this one properly. Should i tackle this next?

**Comment by mountiny** [comment]
Sounds good

**Comment by parasharrajat** [comment]
I have completed the review of the first PR. Once that is merged, we can go to the next ones.

**Comment by mountiny** [comment]
Waiting for next PRs

**Comment by Christinadobrzyn** [comment]
just preparing, how are we going to pay this out, $250 total?

**Comment by parasharrajat** [comment]
I believe it will be more than that. We will have at least 4 PRs. Let's wait for the end. 😄

**Comment by Christinadobrzyn** [comment]
sounds good! Do you have a link to the PRs that we're waiting on so I can track them? Not finding them in a quick review of all the comments.

**Comment by parasharrajat** [comment]
Current PR https://github.com/Expensify/App/pull/63630

**Comment by Christinadobrzyn** [comment]
Thank you @parasharrajat! I'm going to move this back to weekly since that PR is in the works.

**Comment by Christinadobrzyn** [comment]
watching https://github.com/Expensify/App/pull/65245

**Comment by Christinadobrzyn** [comment]
Awesome thank you @parasharrajat - we'll keep this weekly.

**Comment by Christinadobrzyn** [comment]
Not overdue - Melvin

**Comment by Christinadobrzyn** [comment]
montoring https://github.com/Expensify/App/pull/65245

**Comment by chrispader** [comment]
@Christinadobrzyn just updated the PR, awaiting testing and review!

**Comment by Christinadobrzyn** [comment]
monitoring https://github.com/Expensify/App/pull/65245

**Comment by Christinadobrzyn** [comment]
monitoring https://github.com/Expensify/App/pull/65245

**Comment by Christinadobrzyn** [comment]
Just checking.. was this the last PR to deploy before payment/closing? 

https://github.com/Expensify/App/issues/53493#issuecomment-3268688421

**Comment by Christinadobrzyn** [comment]
Thanks @parasharrajat! Melvin keeps moving this back to daily because of the 'due for payment' dates. Moving back to weekly.

**Comment by chrispader** [comment]
> [@chrispader](https://github.com/chrispader) would you be able to lay out the next steps and overview of this project? We are one PR that is going through QA, what else do we plan to have after that?

yes of course!

In terms of the overall modals migrated already, we've really only touched `AttachmentModal` yet. The second 🔴 marked modal `ConfirmModal` which is also causing issues has not been migrated yet.

For the `AttachmentModal`, we **have already merged the [Base PR](https://github.com/Expensify/App/pull/56219)** which implement a new `AttachmentModalScreen` component and 1/6 of the following PRs, which migrate **specific routes/modal types**:

✅ Merged:

- **["Report Attachments" (`ReportAttachmentsModalContent`)](https://github.com/Expensify/App/pull/63630):** The attachment modal used for showing attachments of a report inside the typical carousel view

🚧 Currently in progress:

- **"[Add attachment(s)" (`ReportAddAttachmentModalContent`)](https://github.com/Expensify/App/pull/69523):** Previously the `AttachmentComposerModal` component used to send/upload (multiple) attachments to a report.

The two touched PRs above migrate the most complex routes already, so the remaining routes are easier and more straight-forward...

⏳ Remaining:

The remaining routes/screens are already migrated but i still need to split up **[this previous PR**](https://github.com/Expensify/App/pull/65245) into 4 smaller PRs:

- **"Workspace Avatar" (`WorkspaceAvatarModalContent`)**
- **"Profile Avatar" (`ProfileAvatarModalContent`)**
- **"Report Avatar" (`ReportAvatarModalContent`)**
- **"Transaction Receipt" (`TransactionReceiptModalContent`)**

In terms of the modals marked as 🟡 or 🟢 we would still need to evaluate if these need or should be migrated to "React Navigation modal screens". If we want to do so, since this project has been running for quite a while now, we could think about splitting up the work to speed up things! 🙌🏼

**Comment by Christinadobrzyn** [comment]
just a note for melvin that we're working on this

**Comment by Christinadobrzyn** [comment]
Should we keep this weekly or move it to monthly? @mountiny @chrispader @parasharrajat

**Comment by Christinadobrzyn** [comment]
Just checking, are we still working on all the PRs for this @parasharrajat?

**Comment by Christinadobrzyn** [comment]
@parasharrajat, sorry to bother you so much. Can you confirm the PR that I should follow for payment?

**Comment by chrispader** [comment]
@jmusial let me know if you or somebody needs guidance on the migration, since i have a good understanding of the challenges i faced during the attachment modal migration with web vs. native specific implementations.

If needed, i can also work on this.

**Comment by jmusial** [comment]
> I think we should do that, I am happy to help with creating the issues if you plan it out. [@jmusial](https://github.com/jmusial) would someone from SWM want to take it on?

hey @mountiny  sorry missed that one when I was OOO. Planning on finishing https://github.com/Expensify/App/issues/69171 this week and I think I can work on `SearchRouterModal` after. The rest of the team here in SWM is at capacity rn :(

**Comment by chrispader** [comment]
Okay @mountiny, i'll continue with the follow-ups and use this as the tracking issue.

@jmusial let me know if you need any help with the `SearchRouterModal` migration 🙌🏼

**Comment by mountiny** [comment]
Yeah I dont think we want to change design here

**Comment by parasharrajat** [comment]
Bumping again @chrispader to know ETA for follow-up PRs https://github.com/Expensify/App/issues/53493#issuecomment-3595694267.

**Comment by parasharrajat** [comment]
Sure. No problem.

**Comment by parasharrajat** [comment]
@Christinadobrzyn, I am suggesting creating a new issue for payment of Modal refactors, as we are considering this issue as the main tracking issue for all modals. So, closing this one won't be an option right now. I will post the list of PRs on that issue. Do you agree?

**Comment by parasharrajat** [comment]
@mountiny https://github.com/Expensify/App/issues/53493#issuecomment-3595581393

**Comment by Christinadobrzyn** [comment]
Just checking on this, is there anything to update @parasharrajat?

**Comment by Christinadobrzyn** [comment]
Just a heads up I'm going to be ooo 4/9-4/12.

**Comment by mountiny** [comment]
We might need to find some contributor to help complete this migration

**Comment by jmusial** [comment]
@mountiny what's left to do ? I though we wanted to stop after `SerchRouterModal`  and file uploads ?

**Comment by mountiny** [comment]
This is quite long so I am also out of date here. 

@MelvinBot can you please summarize this issue and where we are with the migration? what modals are still left to be migrated?

**Comment by MelvinBot** [comment]
## Summary of Modal Migration (Issue \#53493)

**Goal:** Replace `react-native-modal`-based modals with `@react-navigation` modal screens to fix animation/navigation issues on native platforms.

### What's been completed

**AttachmentModal migration (🔴 HIGH) — DONE.** All AttachmentModal PRs are merged:
- Base PRs: \#56219, \#69524, \#63630, \#69523
- Route-specific migrations: \#72425, \#72426, \#72427, \#72428, \#72429, \#72430, \#72570
- Cleanup: \#72574
- Follow-ups: \#70740 (attachment/file validation refactor) and \#68551 (multi file upload) — both merged

**SearchRouterModal (🟡 MEDIUM) — DONE.** Migrated via \#79486 by `jmusial` (merged Jan 2026).

### What's still NOT migrated

All remaining modals from the original triage list are **unmigrated**:

**🔴 HIGH (causing animation issues):**
- **ConfirmModal** — on screens like `TagSettingsPage`, `WorkspaceEditTaxPage`, `CategorySettingsPage`, `PolicyDistanceRateDetailsPage` where navigating back while modal animates out causes broken animations

**🟡 MEDIUM (screen-like modals, worth migrating):**
- AmountSelectorModal, AvatarCropModal, CategorySelectorModal, YearPickerModal, TextSelectorModal, ValidateCodeActionModal, ValueSelectorModal, FeatureTrainingModal, PaymentCardCurrencyModal, ConnectToQuickbooksOnlineFlow, ConnectToXeroFlow, CardAuthenticationModal, NetSuiteCustomListSelectorModal, ExpenseLimitTypeSelectorModal, TransactionStartDateSelectorModal, UnitSelectorModal, WorkspaceMemberDetailsRoleSelectionModal, InitialListValueSelectorModal, TypeSelectorModal

**🟢 LOW (contained modals, bottom-to-top, not causing issues):**
- ConfirmModal (non-problematic usages across ~40+ screens), DecisionModal, SelectionListModal, RequireTwoFactorAuthenticationModal, TestToolsModal, SearchSelectedNarrow

### Current status

`chrispader` noted the AttachmentModal migration is complete and follow-up PRs are merged. The `ConfirmModal` (🔴) was identified as the next target but has **not been started**. `mountiny` mentioned potentially finding a contributor to help complete the remaining migrations. `jmusial` asked whether anything beyond SearchRouterModal and file uploads was planned.

**In short:** AttachmentModal and SearchRouterModal are done. The ConfirmModal (🔴 HIGH) is the most important remaining item. All 🟡 MEDIUM and 🟢 LOW modals are untouched.

**Comment by mountiny** [comment]
@jmusial what do you think, should we focus on the confirm modal?

**Comment by chrispader** [comment]
@mountiny @jmusial the `ConfirmModal` should be the only modal that could take some more effort to holistically migrate to a `@react-navigation` modal. This is the only one that is currently causing issues with animation with visible jumps and glitches. For this one we could use a similar solution as with the `AttachmentModal`, though i think it will not be that complex. It would be more important to check ALL usages of it and make sure that it works smoothly across the whole app.

As for the other modals, these should be pretty straightforward to migrate.

1. Create new navigation routes with modal type
2. Extract previous modal code into components that rely on navigation route params instead of React props (params need to be serializable)
3. Update all references of these modals to use navigation instead of `<Modal />`

**Comment by mountiny** [comment]
The question now is if there would be some measurable performance gains from doing this migrations. They might be easy but if there is not much gain, we can deprioritize for other efforts. Do you have any sort of idea about what gains they could have @chrispader ?

**Comment by chrispader** [comment]
These modal migrations are _not_ primarily for performance, **but more about UX and animations**. Currently, we use custom modals everywhere, which have a custom in/out animation, different to the regular screen animation. The problem is that some "modals" to the user just look like any other screen (on mobile),  if the modal animates in from right-to-left, but they will have a different animation than regular React Navigation screens.

The modals marked with `🟢 LOW` are modals that animate in from the bottom, which are not super critical. Here it is mostly fine to use the custom modals. (Except for the `ConfirmModal`, which currently causes major animation issues, seen in the screen recordings)

Modals marked with `🟡 MEDIUM`, these all animate in from right-to-left on mobile, which means they will have a different animation than other regular screens.

So the decision on whether to prioritize or deprioritize is not really about performance gains, but about whether we want to add this better UX and consistency with the rest of the navigation and screen animations. 🙌🏼

**Comment by jmusial** [comment]
I would leave `ConfirmModal` as is. It has been migrated to  async `useConfirmModal` not long ago, that I don't see cost of migrating it would bring reasonable return in perf or smoothness.

IMO 🟡 MEDIUM modals (right-to-left) would be best next step - they look like RHP, and should be them. Before we've been waiting for `dynamic paths` to merge because of `backTo` ban. Now that they're on, modals like `YearPickerModal` should be relative quick wins.

I could try doing one of them, and if they're easy indeed, split up other modals for contributors to pick up ?

**Comment by mountiny** [comment]
Sounds great, thanks @jmusial

**Comment by jmusial** [comment]
Ok so got the [draft PR](https://github.com/Expensify/App/pull/88576) waiting for navigators' review.

Findings:
1. Overall went as expected with testing and finding all the paths being trickiest part. 
2. Some modals look the same but call different actions / require a different param. We need them as separate routes.
2. Sadly `Claude` couldn't do it on his own and getting dynamic paths to work required some manual fixes. 
3. Found a sybling modal under `ROUTES.MONEY_REQUEST_STEP_CATEGORY`. Not sure if it makes sense to unify them?

Proposition with moving forward: 
Go through each one of 🟡 MEDIUM  list and split into 2 lists based on difficulty. Open the `easy` list and manage the `tricky` here. What do you think @mountiny @chrispader ?

**Comment by mountiny** [comment]
yeah open to that approach!

**Comment by jmusial** [comment]
## TLDR

There are four migration categories (plus one "keep" category) based on the modal's state/callback pattern:

- **Group 1 -- API-action selectors:** Selection directly triggers an API call or writes to a known Onyx key. **Migrate directly.**
- **Group 2 -- Lift local `useState` to Onyx:** Selection writes to local `useState` in the parent. Lift that state to Onyx so the page can write it and the parent reads it reactively. **Migrate with state lift.**
- **Group 3 -- Write to known form draft:** The modal is used via `InputWrapper` but in only **one place**, with a known `formID`/`inputID`. A page writes directly to the Onyx form draft, and `FormProvider` merges the change reactively. **Migrate with draft write.**
- **Group 4 -- Complex:** Multiple render sites, platform-specific implementations, complex state (binary data, form reset coupling, OAuth). **Migrate with care or defer.**
- **Group 0 -- Truly generic form-input pickers:** Used in **multiple contexts** with **non-serializable configuration** (callbacks, dynamic items).

## Recommendation

**Group 1** - open to contributors now
**Groups 2, 3** - implement an example, open rest to contributors
**Group 4** - implement here
**Group 0** - Leave as is


 ## TLDR END

---

## Group 0: Keep as Modal (Truly Generic Form-Input Pickers)

These modals are embedded inside **reusable form input components** used via `InputWrapper`/`FormProvider` across **multiple contexts** with **non-serializable configuration** (callbacks, dynamic items). **Keep as-is.**

### TextSelectorModal
- **File:** [`src/components/TextPicker/TextSelectorModal.tsx`](src/components/TextPicker/TextSelectorModal.tsx)
- **Usages:** SamlConfigurationDetails, WorkspaceCreateTaxPage, CreateReportFieldsPage (x2)
- **Why keep:** Uses `FormProvider` internally for validation. Accepts `customValidate` (not serializable), `allowHTML`, `enabledWhenOffline`, `shouldClearOnClose`, and `...rest` text input props. Each of the 4 contexts has different validation logic and API actions.

### ValueSelectorModal
- **File:** [`src/components/ValuePicker/ValueSelectorModal.tsx`](src/components/ValuePicker/ValueSelectorModal.tsx)
- **Usages:** Many form pages via `ValuePicker` (`InputComponent` in `InputWrapper`) + direct use in `ChronosScheduleOOOPage`
- **Why keep:** Dynamic `items` arrays passed as props cannot be serialized to route params. The Chronos direct usage *could* be migrated individually, but the generic `ValuePicker` pattern should remain modal-based.

---

## Group 1: Easy -- Direct API/Onyx Action on Select

One render site, selection triggers a **direct API call or Onyx write**, all data available via route params + Onyx. No state lifting required.

### 1.1 UnitSelectorModal
- **File:** [`src/pages/workspace/distanceRates/UnitSelector/UnitSelectorModal.tsx`](src/pages/workspace/distanceRates/UnitSelector/UnitSelectorModal.tsx)
- **Used by:** Only `UnitSelector/index.tsx` -> `PolicyDistanceRatesSettingsPage.tsx`
- **Action on select:** `onUnitSelected(unit)` -> `setNewUnit` -> **`setPolicyDistanceRatesUnit(...)` API call**

---

## Group 2: Medium -- Lift Local `useState` to Onyx

Single render site, selection writes to **local `useState`** in the parent. Lift to Onyx so the page writes it and the parent reads reactively.

### 2.1 ExpenseLimitTypeSelectorModal
- **File:** [`src/pages/workspace/categories/ExpenseLimitTypeSelector/ExpenseLimitTypeSelectorModal.tsx`](src/pages/workspace/categories/ExpenseLimitTypeSelector/ExpenseLimitTypeSelectorModal.tsx)
- **Used by:** Only `ExpenseLimitTypeSelector.tsx` -> `CategoryFlagAmountsOverPage.tsx`
- **Action on select:** `onExpenseLimitTypeSelected(item.value)` -> `setExpenseLimitType` -> local `useState` consumed by form `onSubmit` calling **`setPolicyCategoryMaxAmount(...)` API**
- **State to lift:** `expenseLimitType` in `CategoryFlagAmountsOverPage` -> `WORKSPACE_CATEGORY_FLAG_AMOUNTS_OVER_FORM` draft or dedicated Onyx key

### 2.2 WorkspaceMemberDetailsRoleSelectionModal
- **File:** [`src/pages/workspace/WorkspaceMemberRoleSelectionModal.tsx`](src/pages/workspace/WorkspaceMemberRoleSelectionModal.tsx)
- **Used by:** Only `ImportedMembersConfirmationPage.tsx`
- **Action on select:** `onRoleChange(item)` -> `setRole(item.value)` local state, consumed by **`importPolicyMembers(policy, membersWithRole)` API**
- **State to lift:** `role` in `ImportedMembersConfirmationPage` -> spreadsheet import form draft or dedicated Onyx key

### 2.3 YearPickerModal
- **File:** [`src/components/DatePicker/CalendarPicker/YearPickerModal.tsx`](src/components/DatePicker/CalendarPicker/YearPickerModal.tsx)
- **Used by:** Only `CalendarPicker/index.tsx`
- **Action on select:** `onYearChange(option.value)` -> updates `currentDateView` state in CalendarPicker
- **State to lift:** `currentDateView` year component from CalendarPicker -> shared Onyx key. Trickiest of the group: CalendarPicker is embedded in `DatePickerModal` and `ScheduleCallPage`, owns complex month/year/day state.

### 2.4 PaymentCardCurrencyModal
- **File:** [`src/components/AddPaymentCard/PaymentCardCurrencyModal.tsx`](src/components/AddPaymentCard/PaymentCardCurrencyModal.tsx)
- **Used by:** Only `PaymentCardChangeCurrencyForm.tsx`
- **Action on select:** `onCurrencyChange(selectedCurrency)` -> `setCurrency(...)` local state, consumed by `changeBillingCurrency(currency, values)` on form submit
- **State to lift:** `currency` in `PaymentCardChangeCurrencyForm` -> `CHANGE_BILLING_CURRENCY_FORM` draft or dedicated Onyx key

---

## Group 3: Medium -- Write to Known Form Draft

Used via `InputWrapper` in only **ONE place** with a known `formID`/`inputID`. Page writes directly to Onyx form draft, `FormProvider` merges reactively.

### 3.1 NetSuiteCustomListSelectorModal
- **File:** [`src/pages/workspace/accounting/netsuite/import/NetSuiteImportCustomFieldNew/NetSuiteCustomListSelectorModal.tsx`](src/pages/workspace/accounting/netsuite/import/NetSuiteImportCustomFieldNew/NetSuiteCustomListSelectorModal.tsx)
- **Used by:** Only `NetSuiteCustomListPicker` -> `ChooseCustomListStep.tsx`
- **Action on select:** `onCustomListSelected(item)` -> `onInputChange(item.value)` + `onInputChange(item.id, internalIDInputID)`
- **State destination:** Form `NETSUITE_CUSTOM_LIST_ADD_FORM`, fields `listName` and `internalID`

### 3.2 InitialListValueSelectorModal
- **File:** [`src/pages/workspace/reports/InitialListValueSelector/InitialListValueSelectorModal.tsx`](src/pages/workspace/reports/InitialListValueSelector/InitialListValueSelectorModal.tsx)
- **Used by:** Only `InitialListValueSelector` -> `CreateReportFieldsPage.tsx`
- **Action on select:** `updateValueInput(initialValue)` -> `onInputChange(value === initialValue ? '' : initialValue)` (toggle behavior)
- **State destination:** Form `WORKSPACE_REPORT_FIELDS_FORM`, field `initialValue`

### 3.3 AmountSelectorModal
- **File:** [`src/components/AmountPicker/AmountSelectorModal.tsx`](src/components/AmountPicker/AmountSelectorModal.tsx)
- **Used by:** Only `AmountPicker` -> `WorkspaceCreateTaxPage.tsx`
- **Action on select:** `onValueSelected(updatedValue)` -> `onInputChange(String(Number(updatedValue)))`
- **State destination:** Form `WORKSPACE_NEW_TAX_FORM`, field `value`

---

## Group 4: Harder -- Complex Migrations

Multiple render sites, platform-specific implementations, complex state management, or fundamentally different UX patterns.

### 4.1 TypeSelectorModal (Report Fields)
- **File:** [`src/pages/workspace/reports/TypeSelector/TypeSelectorModal.tsx`](src/pages/workspace/reports/TypeSelector/TypeSelectorModal.tsx)
- **Used by:** Only `TypeSelector` -> `CreateReportFieldsPage.tsx`
- **Action on select:** `onTypeSelected(reportField.value)` triggers TWO callbacks: (1) `onInputChange` for `type` field, (2) `onTypeSelected` calling `formRef.current?.resetForm({...inputValues, type, initialValue})`
- **State destination:** Form `WORKSPACE_REPORT_FIELDS_FORM`, fields `type` + `initialValue` (plus full form reset)
- **Difficulty:** `resetForm` replaces ALL form values atomically; a draft merge is additive. Type-to-initialValue mapping (date -> `defaultDate`, formula -> `{report:id}`, else -> `''`) must live in the page.
- **Migration approach:** Page writes `type`, `initialValue`, and resets other fields to defaults in one `Onyx.merge`. Alternatively, parent watches draft `type` and triggers `resetForm` reactively.

### 4.2 AvatarCropModal
- **File:** [`src/components/AvatarCropModal/AvatarCropModal.tsx`](src/components/AvatarCropModal/AvatarCropModal.tsx) (471 lines)
- **Used by:** `AvatarWithImagePicker.tsx` (15+ screens) + `AvatarPage.tsx`
- **Difficulty:** Binary image data not serializable to route params. Complex Reanimated state (pan, zoom, rotate). `cropOrRotateImage` returns `File | CustomRNImageManipulatorResult`. `maskImage` differs between usage sites.
- **Migration approach:** Store picked image URI in temporary Onyx key -> crop page reads/crops/writes back -> parent reads result and calls upload API.

### 4.3 FeatureTrainingModal
- **File:** [`src/components/FeatureTrainingModal.tsx`](src/components/FeatureTrainingModal.tsx) (534 lines)
- **Used by:** 11+ files, 12+ render sites. 3 already navigation pages via `FeatureTrainingModalNavigator`; rest are `CENTERED` overlays.
- **Difficulty:** Generic educational overlay with video/SVG/Lottie, "don't show again", help/skip buttons, custom children. Not an RHP pattern.
- **Migration:** **Recommend deferring.** Inline usages (HoldOrReject, HoldSubmitter, multi-scan popup) are contextual overlays that would change UX.

### 4.4 ConnectToQuickbooksOnlineFlow / ConnectToXeroFlow
- **Files:** [`src/components/ConnectToQuickbooksOnlineFlow/`](src/components/ConnectToQuickbooksOnlineFlow/), [`src/components/ConnectToXeroFlow/`](src/components/ConnectToXeroFlow/)
- **Used by:** Only via `getAccountingIntegrationData` -> `AccountingContext`
- **Difficulty:** Platform-specific (web: external link / native: WebView modal). Xero has 2FA branching. QBO disables taxes on mount. Imperatively triggered via `startIntegrationFlow`.
- **Migration approach:** Create dedicated pages (QBOConnectPage, XeroConnectPage). Navigate instead of mounting via `setupConnectionFlow`. Decouple from `AccountingContext`.

### 4.5 CardAuthenticationModal
- **File:** [`src/pages/settings/Subscription/CardAuthenticationModal/index.tsx`](src/pages/settings/Subscription/CardAuthenticationModal/index.tsx)
- **Used by:** `PaymentCard/index.tsx` (no policyID), `WorkspaceOwnerChangeWrapperPage.tsx` (with policyID)
- **Difficulty:** Web-only (native returns `null`). Visibility driven by Onyx (`VERIFY_3DS_SUBSCRIPTION`). `window.addEventListener('message')` for SCA 3D Secure. Different success actions based on `policyID`.
- **Migration approach:** Navigation page reading `VERIFY_3DS_SUBSCRIPTION`, navigate when `authenticationLink` appears, pass `policyID` as route param.

---

## Not Applicable

### ValidateCodeActionModal
Already navigation-based. Dead type `ValidateCodeActionModalProps` in [`src/components/ValidateCodeActionModal/type.ts`](src/components/ValidateCodeActionModal/type.ts) can be cleaned up.

### TransactionStartDateSelectorModal
Does not exist. Already implemented as navigation pages (`TransactionStartDateStep.tsx`, `PersonalCardEditTransactionStartDatePage.tsx`, `WorkspaceCompanyCardEditTransactionStartDatePage.tsx`).

**Comment by parasharrajat** [comment]
Reviewed PR - To Track. 

- https://github.com/Expensify/App/pull/77380
- https://github.com/Expensify/App/pull/70740
- https://github.com/Expensify/App/pull/68551 - Caused a Regression (Rework https://github.com/Expensify/App/pull/84423)
- https://github.com/Expensify/App/pull/88576
- https://github.com/Expensify/App/pull/88915

**Comment by jmusial** [comment]
hey @parasharrajat  https://github.com/Expensify/App/pull/88316 belongs to a different issue I belive

**Comment by parasharrajat** [comment]
Fixed. Thanks. @jmusial

**Comment by parasharrajat** [comment]
In progress. There will be more PRs.

**Comment by mhawryluk** [comment]
I am working on migrating one of the modals from Group 3 (AmountSelectorModal)

**Comment by mountiny** [comment]
@mhawryluk @jmusial I would like to create sub issues so we dont link it all to the tracking, can you help me out with laying out the next phases and buckets for the migration so I can create these sub issues reasonably?

**Comment by mallenexpensify** [comment]
> I would like to create sub issues 

Is @parasharrajat due payment here?

**Comment by mountiny** [comment]
@parasharrajat yeah I think if you could summarize what was not paid out yet, we can pay out and hten move the other task to the sub issues

**Comment by jmusial** [comment]
> [@mhawryluk](https://github.com/mhawryluk) [@jmusial](https://github.com/jmusial) I would like to create sub issues so we dont link it all to the tracking, can you help me out with laying out the next phases and buckets for the migration so I can create these sub issues reasonably?

Basing on [this comment](https://github.com/Expensify/App/issues/53493#issuecomment-4312826488): 

### Create as subtask and open to contributors: 

Mentioning above comment and ref PRs as a reference in sub issues.

- 1.1 UnitSelectorModal

[Reference PR for 2](https://github.com/Expensify/App/pull/88915)
- 2.2 WorkspaceMemberDetailsRoleSelectionModal
- 2.3 YearPickerModal
- 2.4 PaymentCardCurrencyModal

[ReferencePR for 3](https://github.com/Expensify/App/pull/90346)
- 3.1 NetSuiteCustomListSelectorModal
- 3.2 InitialListValueSelectorModal

### Create as subtask and we'll implement:

Mentioning above comment as a reference in sub issues.

- 4.1 TypeSelectorModal (Report Fields)
- 4.2 AvatarCropModal
- 4.3 FeatureTrainingModal
- 4.4 ConnectToQuickbooksOnlineFlow / ConnectToXeroFlow
- 4.5 CardAuthenticationModal

**Comment by parasharrajat** [comment]
@mallenexpensify @mountiny https://github.com/Expensify/App/issues/53493#issuecomment-4323956496 This is the list of completed PRs.

**Comment by mountiny** [comment]
@jmusial thanks!

Created the issues, who will take these:
https://github.com/Expensify/App/issues/90473
https://github.com/Expensify/App/issues/90474
https://github.com/Expensify/App/issues/90475
https://github.com/Expensify/App/issues/90476
https://github.com/Expensify/App/issues/90477

**Comment by parasharrajat** [comment]
@mountiny I can pick contributors on it. Correct? or they will be solved experts.

**Comment by mountiny** [comment]
I think these 5 would be solved by SWM

**Comment by jmusial** [comment]
@mountiny you can assign them to me for now, I'll try to find someone to do them

**Comment by parasharrajat** [comment]
@mallenexpensify I am due for 1125 here as per https://github.com/Expensify/App/issues/53493#issuecomment-4323956496

**Comment by jmusial** [comment]
Not overdue, the migration effort is goiing through sub-issues.

**Comment by mountiny** [comment]
Tracking issue

**Comment by mallenexpensify** [comment]
### Payment Summary
Contributor+: @parasharrajat  due $1125 via NewDot

Breakdown [here](https://github.com/Expensify/App/issues/53493#issuecomment-4323956496)

**Comment by jmusial** [comment]
Hey @mountiny, @borys3kk will be handling those: 

https://github.com/Expensify/App/issues/90474
https://github.com/Expensify/App/issues/90475
https://github.com/Expensify/App/issues/90476
https://github.com/Expensify/App/issues/90477

@borys3kk - the issues should have the context but just in case [this comment](https://github.com/Expensify/App/issues/53493#issuecomment-4312826488) has the whole picture.

**Comment by parasharrajat** [comment]
Payment requested as per https://github.com/Expensify/App/issues/53493#issuecomment-4490899468

**Comment by garrettmknight** [comment]
$1125 approved for @parasharrajat

**Comment by parasharrajat** [comment]
Lets' move it weekly.... It is a tracking issue now.

**Comment by mountiny** [comment]
Not overdue

**Comment by mountiny** [comment]
Chugging through the sub issues

**Comment by mountiny** [comment]
Working through this

**Comment by mountiny** [comment]
underway

**Comment by mountiny** [comment]
Almost there

**Comment by mountiny** [comment]
@jmusial I believe we are close to being done, mind summarizing if we have anything left to do?

**Comment by mountiny** [comment]
Working on https://github.com/Expensify/App/pull/97172

**Comment by chrispader** [comment]
@MelvinBot please unassign me from this as i'm not working on this anymore.

**Comment by MelvinBot** [comment]
🤖 Done — unassigned `chrispader` from this issue.

Remaining assignees: `jmusial`, `mallenexpensify`, `parasharrajat`, `mountiny`.


---
_[view run](https://github.com/Expensify/melvin/actions/runs/34835807185)_ · _no recording available_
